yt_dlp/extractor/vimeo.py

```diff
@@ -253,6 +253,23 @@ def _extract_original_format(self, url, video_id, unlisted_hash=None):
                             'quality': 1,
                         }
 
+    def _extract_original_format_2(self, video_id):
+        jwt_response = self._download_json('https://vimeo.com/_rv/viewer', video_id)
+        if jwt_response.get('jwt'):
+            headers = {'Authorization': f"jwt {jwt_response['jwt']}"}
+            original_response = self._download_json('https://api.vimeo.com/videos/' + video_id, video_id, headers=headers)
+            for download_data in original_response.get('download'):
+                if download_data['quality'] == 'source':
+                    return {
+                        'url': download_data['link'],
+                        'ext': download_data['link'][-3:] or 'mp4',
```

Comment:
idk of any 4 letter extension for a video file.
Lemme know how this should be handled 
`determine_ext` this wasn't working

Reply:
This should be more robust now considering how the api gives url.